### PR TITLE
python37Packages.scikitlearn: patching build

### PR DESCRIPTION
**Pull request by d-goldin**
###### Motivation for this change

For numpy>=1.17 a test-case broke that required adjustments to
a threshold.

See https://github.com/NixOS/nixpkgs/issues/68494
Related: https://github.com/NixOS/nixpkgs/issues/68361

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @FRidh due to previous discussion of this

### CODE REVIEW

File: pkgs/development/python-modules/scikitlearn/default.nix [hunk 1]
```diff
@@ -40,6 +41,15 @@ buildPythonPackage rec {
   ];
   checkInputs = [ pytest ];
 
+  patches = [
+    # Fixes tests by changing threshold of a test-case that broke
+    # with numpy versions >= 1.17. This should be removed for versions > 0.21.2.
+	( fetchpatch {
+	  url = "https://github.com/scikit-learn/scikit-learn/commit/b730befc821caec5b984d9ff3aa7bc4bd7f4d9bb.patch";
+	  sha256 = "0i621flcfqgjcw37p96zx7hfrkghlvd88wi816i4kbs0gxg0vm6p";
```
Comment (marsam) [inline pkgs/development/python-modules/scikitlearn/default.nix]: I'm getting a different hash
```suggestion
	  sha256 = "0z36m05mv6d494qwq0688rgwa7c4bbnm5s2rcjlrp29fwn3fy1bv";
```

Comment (d-goldin) [reply pkgs/development/python-modules/scikitlearn/default.nix]: Hm, let me double check. Likely I messed it up locally.

Comment (d-goldin) [reply pkgs/development/python-modules/scikitlearn/default.nix]: Fixed, thanks.

**Comment by marsam** [comment]
thank you @d-goldin

**Pull request merged**